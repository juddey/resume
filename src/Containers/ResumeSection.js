import React from 'react'
import PropTypes from 'prop-types'
import { Hidden, Grid } from '@material-ui/core'
// Styles
import styles from './Styles/ResumeSectionStyle'
import { Colors } from '../Themes/'

function ResumeSection (props) {
  return (
    <Grid
      container
      justify='center'
      style={{ backgroundColor: props.backgroundColor }}
    >
      <Hidden only='xs'>
        <Grid item />
      </Hidden>
      <Grid
        item
        xs={props.gridSpread[0]}
        sm={props.gridSpread[1]}
        md={props.gridSpread[2]}
        lg={props.gridSpread[3]}
        xl={props.gridSpread[4]}
        style={{ ...styles.middleGrid, ...props.middleGrid }}
      >
        <div style={styles.sectionTitle}>{props.title}</div>
        {props.section}
      </Grid>
      <Hidden only='xs'>
        <Grid item />
      </Hidden>
    </Grid>
  )
}
ResumeSection.defaultProps = {
  backgroundColor: Colors.background,
  gridSpread: [12, 10, 8, 6, 4]
}

ResumeSection.propTypes = {
  backgroundColor: PropTypes.string,
  gridSpread: PropTypes.array,
  middleGrid: PropTypes.object
}

export default ResumeSection
