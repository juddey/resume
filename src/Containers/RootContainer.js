import React, { Component } from 'react'
import Navigation from '../Navigation/AppNavigation'
import { connect } from 'react-redux'
import StartupActions from '../Redux/StartupRedux'
import ReduxPersist from '../Config/ReduxPersist'
import { IntlProvider, addLocaleData } from 'react-intl'
import en from 'react-intl/locale-data/en'
import enTranslationMessages from '../Translations/en.json'

addLocaleData([...en])

export const appLocales = ['en']

export const DEFAULT_LOCALE = 'en'

export const formatTranslationMessages = (locale, messages) => {
  const defaultFormattedMessages =
    locale !== DEFAULT_LOCALE
      ? formatTranslationMessages(DEFAULT_LOCALE, enTranslationMessages)
      : {}
  return Object.keys(messages).reduce((formattedMessages, key) => {
    const formattedMessage =
      !messages[key] && locale !== DEFAULT_LOCALE
        ? defaultFormattedMessages[key]
        : messages[key]
    return Object.assign(formattedMessages, { [key]: formattedMessage })
  }, {})
}

export const translationMessages = {
  en: formatTranslationMessages('en', enTranslationMessages)
  // You can add other languages here.
}

class RootContainer extends Component {
  componentDidMount () {
    // if redux persist is not active fire startup action
    if (!ReduxPersist.active) {
      this.props.startup()
    }
  }

  render (messages) {
    return (
      <IntlProvider locale='en' messages={translationMessages.en}>
        <Navigation />
      </IntlProvider>
    )
  }
}

// wraps dispatch to create nicer functions to call within our component
const mapDispatchToProps = dispatch => ({
  startup: () => dispatch(StartupActions.startup())
})

export default connect(null, mapDispatchToProps)(RootContainer)
