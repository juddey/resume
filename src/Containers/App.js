import React, { Component } from 'react'
import { Provider } from 'react-redux'
import RootContainer from './RootContainer'
import createStore from '../Redux'

// create our store
const store = createStore()

/**
 * Provides an entry point into our application.
 *
 * We create our Redux store here, put it into a provider and then bring in our
 * RootContainer.
 *
 */
class App extends Component {
  render () {
    return (
      <Provider store={store}>
        <RootContainer />
      </Provider>
    )
  }
}

export default App
