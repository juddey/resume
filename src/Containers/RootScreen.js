import React from 'react'
import {
  Avatar,
  About,
  IconBar,
  Interests,
  ResumeTitle,
  SectionTitle,
  Skills,
  ExperienceCard
} from '../Components'
import { Grid, Typography } from '@material-ui/core'
import ExperienceData from '../data/Experience'
import { ResumeSection } from '../Containers'
import { Colors } from '../Themes'
import styles from './Styles/RootScreenStyle'

function RootScreen () {
  return (
    <div>
      <Grid container style={styles.container}>
        <div style={styles.heading}>
          <div style={{ marginBottom: '10px' }}>
            <ResumeTitle />
          </div>
          <Typography
            variant='subheading'
            style={{
              letterSpacing: '1px',
              color: 'white',
              marginBottom: '10px'
            }}
          >
            software developer
          </Typography>
          <IconBar />
        </div>
        <ResumeSection
          title={<Avatar />}
          section={<About />}
          backgroundColor={Colors.white}
          gridSpread={[10, 10, 8, 6, 4]}
          middleGrid={{
            display: 'flex',
            flex: 1,
            alignItems: 'center',
            flexDirection: 'column'
          }}
        />
      </Grid>
      <ResumeSection
        title={<SectionTitle title='Experience' />}
        section={ExperienceData.map((data, i) => (
          <div key={i} style={{ marginBottom: 10 }}>
            <ExperienceCard
              headerLeftText={data.position}
              headerRightText={data.duration}
              cardText={data.cardText}
              chips={data.chips}
              avatarImage={data.avatarImage}
            />
          </div>
        ))}
        backgroundColor={Colors.lightGray}
        gridSpread={[11, 10, 8, 6, 4]}
      />
      <ResumeSection
        title={<SectionTitle title='Skills' />}
        section={<Skills />}
        backgroundColor={Colors.white}
      />
      <ResumeSection
        title={<SectionTitle title='Interests' color='white' />}
        section={<Interests />}
      />
    </div>
  )
}

export default RootScreen
