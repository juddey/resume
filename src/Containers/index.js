// Container Index
import ResumeSection from './ResumeSection'
import RootScreen from './RootScreen'

// prettier-ignore
export {
  ResumeSection,
  RootScreen
}
