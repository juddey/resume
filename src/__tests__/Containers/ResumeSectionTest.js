import React from 'react'
import configureStore from 'redux-mock-store'
import { createComponentWithIntl } from '../../../src/Support/Utils'

import { ResumeSection } from '../../Containers/'
// import { INITIAL_STATE } from '../../Redux/YourNameRedux'
const mockStore = configureStore()

const initialState = {}
const store = mockStore(initialState)

test('renders to snapshot', () => {
  const tree = createComponentWithIntl(
    <ResumeSection store={store} {...initialState} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
