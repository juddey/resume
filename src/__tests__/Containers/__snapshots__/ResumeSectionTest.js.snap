// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders to snapshot 1`] = `
<div
  className="MuiGrid-container-1 MuiGrid-justify-xs-center-18"
  style={
    Object {
      "backgroundColor": "#012C4C",
    }
  }
>
  <div
    className="MuiGrid-item-2"
  />
  <div
    className="MuiGrid-item-2 MuiGrid-grid-xs-12-39 MuiGrid-grid-sm-10-50 MuiGrid-grid-md-8-61 MuiGrid-grid-lg-6-72 MuiGrid-grid-xl-4-83"
    style={
      Object {
        "paddingBottom": "50px",
        "paddingTop": "50px",
      }
    }
  >
    <div
      style={
        Object {
          "display": "flex",
          "justifyContent": "center",
          "marginBottom": "30px",
        }
      }
    />
  </div>
  <div
    className="MuiGrid-item-2"
  />
</div>
`;
